delfin/drivers/hitachi/vsp/rest_handler.py

```diff
@@ -0,0 +1,208 @@
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+import threading
+
+import requests
+import six
+from oslo_log import log as logging
+
+from delfin import cryptor
+from delfin import exception
+from delfin.drivers.hitachi.vsp import consts
+
+LOG = logging.getLogger(__name__)
+STATUS_200 = 200
+STATUS_201 = 201
+STATUS_202 = 202
+STATUS_204 = 204
+STATUS_401 = 401
+
+
+class RestHandler(object):
+    HDSVSP_SYSTEM_URL = '/ConfigurationManager/v1/objects/storages'
+    HDSVSP_LOGOUT_URL = '/ConfigurationManager/v1/objects/sessions/'
+    HDSVSP_COMM_URL = '/ConfigurationManager/v1/objects/storages/'
+
+    HDSVSP_AUTH_KEY = 'Authorization'
+
+    def __init__(self, rest_client):
+        self.rest_client = rest_client
+        self.session_lock = threading.Lock()
+        self.hdsvsp_session_id = None
+        self.storage_device_id = None
+        self.device_model = None
+        self.serial_number = None
+
+    def call(self, url, data=None, method=None):
+        try:
+            res = self.rest_client.do_call(url, data, method,
+                                           calltimeout=consts.SOCKET_TIMEOUT)
+            if (res.status_code == consts.ERROR_SESSION_INVALID_CODE
+                    or res.status_code ==
+                    consts.ERROR_SESSION_IS_BEING_USED_CODE):
+                LOG.error("Failed to get token=={0}=={1},get token again"
+                          .format(res.status_code, res.text))
+                # if method is logout,return immediately
+                if method == 'DELETE' and RestHandler. \
+                        HDSVSP_LOGOUT_URL in url:
+                    return res
+                self.rest_client.rest_auth_token = None
+                access_session = self.login()
+                if access_session is not None:
+                    res = self.rest_client. \
+                        do_call(url, data, method,
+                                calltimeout=consts.SOCKET_TIMEOUT)
+                else:
+                    LOG.error('Login error,get access_session failed')
+            elif res.status_code == 503:
+                raise exception.InvalidResults(res.text)
+
+            return res
+
+        except Exception as e:
+            err_msg = "Get RestHandler.call failed: %s" % (six.text_type(e))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def get_resinfo_call(self, url, data=None, resName=None):
+        result_json = None
+        res = self.call(url, data, 'GET')
+        if res is not None:
+            if res.status_code == STATUS_200:
```

Comment:
Why always set 200 to `status_code` manually? What's the original value of res.status_code?

Reply:
changed